pkg/iter/cache.go

```diff
@@ -35,43 +36,52 @@ func (it *cachedIterator) Reset() {
 	it.curr = -1
 }
 
-func (it *cachedIterator) Next() bool {
-	if it.base != nil {
-		ok := it.base.Next()
-		// we're done with the base iterator.
-		if !ok {
-			it.closeErr = it.base.Close()
-			it.iterErr = it.base.Error()
-			it.base = nil
-			return false
-		}
-		// we're caching entries
-		it.cache = append(it.cache, entryWithLabels{entry: it.base.Entry(), labels: it.base.Labels()})
-		it.curr++
-		return true
+func (it *cachedIterator) Base() EntryIterator {
+	return it.base
+}
+
+func (it *cachedIterator) consumeNext() bool {
+	if it.base == nil {
+		return false
+	}
+	ok := it.base.Next()
```

Comment:
Should we not be using the `Base()` function instead of these direct member accesses?

Reply:
Might give us some leverage later, since we have this layer of indirection.